Dockerfile

```diff
@@ -0,0 +1,103 @@
+        libtiff-dev         \
+        libopenjp2-7-dev    \
+        libavformat-dev     \
+        libssl-dev          \
+        libpq-dev           \
+        libeigen3-dev       \
+        libgl1-mesa-dev     \
+        libboost1.67-all-dev\
+        libglew-dev         \
+        qtbase5-dev         \
+        libqt5x11extras5-dev\
+        libqt5opengl5-dev   \
+        libsuitesparse-dev  \
+        meshlab
+
+RUN wget https://www.python.org/ftp/python/3.9.9/Python-3.9.9.tgz
+RUN tar xzvf Python-3.9.9.tgz
+WORKDIR Python-3.9.9
+RUN ./configure
+RUN make install
+WORKDIR /
+
+RUN python3 -m pip install numpy \
+    && wget -q https://github.com/opencv/opencv/archive/${OPENCV_VERSION}.zip -O opencv.zip \
+    && unzip -qq opencv.zip -d /opt \
+    && rm -rf opencv.zip \
+    && cmake \
+        -D BUILD_TIFF=ON            \
+        -D BUILD_opencv_java=OFF    \
+        -D WITH_CUDA=OFF            \
+        -D WITH_OPENGL=ON           \
+        -D WITH_OPENCL=OFF          \
+        -D WITH_IPP=ON              \
+        -D WITH_TBB=ON              \
+        -D WITH_EIGEN=ON            \
+        -D WITH_V4L=ON              \
+        -D BUILD_TESTS=OFF          \
+        -D BUILD_PERF_TESTS=OFF     \
+        -D CMAKE_BUILD_TYPE=RELEASE \
+        -D CMAKE_INSTALL_PREFIX=$(python3.9 -c "import sys; print(sys.prefix)") \
+        -D PYTHON_EXECUTABLE=$(which python3.9) \
+        -D PYTHON_INCLUDE_DIR=$(python3.9 -c "from distutils.sysconfig import get_python_inc; print(get_python_inc())") \
+        -D PYTHON_PACKAGES_PATH=$(python3.9 -c "from distutils.sysconfig import get_python_lib; print(get_python_lib())") \
+        /opt/opencv-${OPENCV_VERSION} \
+    && make -j$(nproc) \
+    && make install \
+    && rm -rf /opt/build/* \
+    && rm -rf /opt/opencv-${OPENCV_VERSION} \
+    && rm -rf /var/lib/apt/lists/* \
+    && apt-get -qq autoremove \
+    && apt-get -qq clean
+
+RUN git clone https://github.com/dorian3d/DLib
+WORKDIR DLib
+RUN git reset --hard b6c28fb
+RUN mkdir build
+WORKDIR build
+RUN cmake -E env CXXFLAGS="-march=native" cmake .. && make install
+WORKDIR /
+
+RUN git clone https://github.com/laurentkneip/opengv.git
+WORKDIR opengv
+RUN git reset --hard 91f4b19
+RUN mkdir build
+WORKDIR build
+RUN cmake -E env CXXFLAGS="-march=native" cmake .. && make -j install
+WORKDIR /
+
+RUN git clone https://github.com/RainerKuemmerle/g2o.git
+WORKDIR g2o
+RUN git reset --hard b1ba729
+RUN mkdir build
+WORKDIR build
+RUN cmake -DBUILD_WITH_MARCH_NATIVE=ON .. && make -j install
+
+WORKDIR /datasets
+RUN wget https://www.eth3d.net/data/slam/datasets/einstein_1_mono.zip
+RUN wget https://www.eth3d.net/data/slam/datasets/einstein_1_rgbd.zip
+RUN unzip einstein_1_mono.zip
+RUN unzip einstein_1_rgbd.zip
+
+ADD . /home/badslam
+RUN mkdir build_RelWithDebInfo
+WORKDIR /home/badslam/build_RelWithDebInfo
+
+RUN cmake -E env CXXFLAGS="-march=native" cmake -DCMAKE_BUILD_TYPE=RelWithDebInfo -DCMAKE_CUDA_FLAGS="-arch=sm_61" .. && make -j
```

Comment:
`-DCMAKE_CUDA_FLAGS="-arch=sm_61"` probably shouldn't be hardcoded here, or at least the Readme should mention that it may need to be adapted to the user's GPU before building the Docker image. Since I am not familiar with Docker, I don't know whether there is a good way to pass parameters to the build.